requirements/tools.txt

```diff
@@ -1,3 +1,3 @@
 mypy==1.11.2
 flake8==6.0.0
-black==24.8.0             # Until we drop Python 3.6 support, we have to stay with this version
+black==24.10.0             # Until we drop Python 3.6 support, we have to stay with this version
```

Comment:
@WilliamBergamin @seratch Dependabot would like to bump us to `black@24.10.0` from `24.8.0`. The Release Notes mention dropping Python 3.8 support and the comment above mentions that we still support Python 3.6. So, it sounds like the latest version of black may not run reliably on Python 3.6.

What's your advice on this PR?

Reply:
@mwbrooks Thanks for checking this. Upgrading black to a newer version does not bring any benefits and I personally think still we should keep 3.6 support for a while (say, one or two more years). It's indeed EOLed a long time ago! but for Python and Java, providing much longer supports for EOLed versions would be expected in many cases.

Reply:
Thanks @seratch! Is there a range syntax that we can use to prevent @dependabot from trying to upgrade it? For example `black<=24.8.0`?

Reply:
>Run all the unit tests / build (3.6) (pull_request)

The test execution does not fail (because it does not use black) but let's keep the current black version this time

Reply:
>Thanks @seratch! Is there a range syntax that we can use to prevent https://github.com/dependabot from trying to upgrade it? For example black<=24.8.0?

Hmm, honestly i am not sure about the way to prevent this. Even if you use a version range, the bot still tries to upgrade to newer range 😅

Reply:
Ah, alright. Well, the comment left on the dependency was enough for me to question accepting the upgrade. Hopefully it'll be enough for other reviewers to know that we don't want to upgrade it yet.